src/custom/controller.go

```diff
@@ -0,0 +1,177 @@
+package custom
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	appsinformers "k8s.io/client-go/informers/apps/v1"
+	"k8s.io/client-go/kubernetes"
+	appslisters "k8s.io/client-go/listers/apps/v1"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/util/workqueue"
+)
+
+type controller struct {
+	clientSet      kubernetes.Interface
+	depLister      appslisters.DeploymentLister
+	depCacheSynced cache.InformerSynced
+	workQueue      workqueue.RateLimitingInterface
+}
+
+func (cntrlreceiver *controller) handleAdd(obj interface{}) {
+	cntrlreceiver.workQueue.Add(obj)
+}
+
+func (cntrlreceiver *controller) Run(stopCh <-chan struct{}) {
+	fmt.Println("Starting Custom Controller....")
+	if !cache.WaitForCacheSync(stopCh, cntrlreceiver.depCacheSynced) {
+		fmt.Println("Waiting for the cache to be synced....")
+	}
+
+	go wait.Until(cntrlreceiver.worker, 1*time.Second, stopCh)
+
+	<-stopCh
+}
+
+func (cntrlreceiver *controller) worker() {
+	for cntrlreceiver.processItem() {
+	}
+}
+
+func (cntrlreceiver *controller) processItem() bool {
+	item, shutdown := cntrlreceiver.workQueue.Get()
+	if shutdown {
+		return false
+	}
+
+	defer cntrlreceiver.workQueue.Forget(item)
+
+	key, err := cache.MetaNamespaceKeyFunc(item)
+	if err != nil {
+		fmt.Println("Error in cache.MetaNamespaceKeyFunc(): ", err.Error())
+	}
+
+	ns, name, err := cache.SplitMetaNamespaceKey(key)
+	if err != nil {
+		fmt.Println("Error in cache.SplitMetaNamespaceKey(): ", err.Error())
+	}
+
+	err = cntrlreceiver.createSecret(ns, name)
+	if err != nil {
+		fmt.Println("Error in cntrlreceiver.createSecrets(): ", err.Error())
+		return false
+	}
+	return true
+}
+
+func (cntrlreceiver *controller) createSecret(ns, name string) error {
```

Comment:
why are we creating the secret, I don't remember specifying that user will not create the secret.